### ISSUE TITLE
Make idempotent

### ISSUE THREAD

**Issue by websocket98765** [svelte-add/tailwindcss#17]
It'd be ideal if this checked whether it was run previously. Easiest way would probably be to check if `postcss.config.cjs` exists and already contains `tailwindcss`, and to abort the script if it exists. 

That would prevent it acting twice on the same code base, which will break a project until postcss.config.cjs is cleaned up again, overwrites a user's custom tailwind config, etc. (I intentionally ran the command twice to check if it was idempotent, with the below result for `postcss.config.cjs`).

```
const tailwindcss = require("@tailwindcss/jit");
const tailwindcss = require("@tailwindcss/jit");
const autoprefixer = require("autoprefixer");
const cssnano = require("cssnano");

const mode = process.env.NODE_ENV;
const dev = mode === "development";

module.exports = {
	plugins: [
		// Some plugins, like postcss-nested, need to run before Tailwind
		
		tailwindcss,
		
		// But others, like autoprefixer, need to run after

		// Some plugins, like postcss-nested, need to run before Tailwind
		
		tailwindcss,
		
		// But others, like autoprefixer, need to run after

		autoprefixer,
		
		!dev && cssnano({
			preset: "default",
		}),
	],
};
```

**Comment by babichjacob** [comment]
This is definitely not a bad idea, but I don't think it brings enough benefit to prioritize implementing it over anything else.

**Comment by babichjacob** [comment]
Don't be confused by the `wontfix` label. It just signals that I can't fix this in the current state of things. Wait like 2-3 weeks for the rewrite (I'm really busy for the next week).

This will be solved in a future rewrite of `svelte-add`: https://github.com/svelte-add/svelte-adders/issues/12

Specifically, it won't be allowed to run `svelte-add` after the project initialization process https://github.com/svelte-add/svelte-adders/issues/17 so idempotency won't be needed.

**Comment by websocket98765** [comment]
Not ideal. People may start a project and later decide they want TailwindCSS, etc. But I'm sure you're aware of this

**Comment by babichjacob** [comment]
> 
> 
> Not ideal. People may start a project and later decide they want TailwindCSS, etc. But I'm sure you're aware of this

It is almost quite literally impossible for an automated tool to solve this. People can follow tutorials or replicate what this tool does themselves.

**Comment by websocket98765** [comment]
No, it's really not. https://www.ansible.com solves it. Even NextJS's Tailwind `npx` command runs checks to see if Tailwind has already been installed. I also wrote JS code showing how such checks could be achieved in JS too in a previous issue on this repo. 

I think we just have a difference of opinion on how a tool like this should behave: I'm looking for idempotency and expect such a tool to not add demo cruft or to alter any files in the project except those needed to achieve its goal. It was fun to playing with this--thanks for writing it. I'll probably end up writing my own scripts, it sounds like. Cheers.

**Comment by babichjacob** [comment]
Idempotency is the easy one, but supporting modifying code that we don't know anything about is practically impossible.

Excluding demos has been documented for a few weeks: https://github.com/svelte-add/svelte-add#-excluding-examples

**Comment by babichjacob** [comment]
> 
> 
> > Not ideal. People may start a project and later decide they want TailwindCSS, etc. But I'm sure you're aware of this
> 
> It is almost quite literally impossible for an automated tool to solve this. People can follow tutorials or replicate what this tool does themselves.

@websocket98765 I relent. It'll still be possible (will be made more possible than the current state of things) to run the `svelte-add` command at a later time than project initialization.

There will be a warning message to inform that if it doesn't work, the issue should be reported and if they want instant results they should run the project initializer again.

(Reminder not to be confused by the `wontfix` label -- it just signals that the issue cannot be solved until the rewrite, which can be tracked here: https://github.com/orgs/svelte-add/projects/1#card-59584691)
